src/smolagents/models.py

```diff
@@ -304,6 +305,14 @@ def _prepare_completion_kwargs(
                 }
             )
 
+        # The Bedrock API does not support the `type` key in requests.  
+        # This block of code modifies the object to meet Bedrock's requirements.
+        if is_bedrock_runtime:
+            for message in completion_kwargs.get("messages", []):
+                for content in message.get("content", []):
+                    if "type" in content:
+                        del content["type"]
+
```

Comment:
What if you override the `_prepare_completion_kwargs` in `AmazonBedrockServerModel` and call super() and add the new functionality there?

Reply:
Thank you so much for this feedback! Looking at other models, I could understand the use of `_prepare_completion_kwargs` and it makes a lot of sense that I have an override to implement Bedrock's logic. The code is much clearer now.